lang/ir/src/ir/item/storage.rs

```diff
@@ -91,16 +91,17 @@ impl TryFrom<syn::ItemStruct> for Storage {
                 "generic ink! storage structs are not supported",
             ))
         }
-        match &item_struct.vis {
-            syn::Visibility::Inherited
-            | syn::Visibility::Restricted(_)
-            | syn::Visibility::Crate(_) => {
-                return Err(format_err_spanned!(
-                    &item_struct.vis,
-                    "non `pub` ink! storage structs are not supported",
-                ))
-            }
-            _ => (),
+        let bad_visibility = match &item_struct.vis {
```

Comment:
Duplicate of code in `event.rs`

Reply:
resolved. thanks for pointing out! duplicate code often happens during prototyping of new features etc. so this is super helpful!